test/credentials.spec.js

```diff
@@ -1080,334 +1152,191 @@
-          }
-        );
-
-        it(
-          'throws an error when the full URI environment variable contains a URI with an unsupported protocol',
-          function () {
-            process.env['AWS_CONTAINER_CREDENTIALS_FULL_URI'] = 'http://s3-us-west-2.amazonaws.com/bucket/credentials';
-            expect(creds.getECSFullUri.bind(creds))
-              .to.throw(/Unsupported hostname/);
-          }
-        );
-
-        it(
-          'returns a full URI when the full URI environment variable is set to a non-localhost https URI',
-          function () {
-            process.env['AWS_CONTAINER_CREDENTIALS_FULL_URI'] = 'https://s3-us-west-2.amazonaws.com/bucket/credentials';
-            expect(creds.getECSFullUri())
-              .to.equal('https://s3-us-west-2.amazonaws.com/bucket/credentials');
-          }
-        );
-      });
-
-      describe('needsRefresh', function() {
-        return it('can be expired based on expire time from URI endpoint', function() {
-          var spy;
-          process.env['AWS_CONTAINER_CREDENTIALS_RELATIVE_URI'] = '/path';
-          spy = mockEndpoint(new Date(0));
-          creds.refresh(function() {});
-          expect(spy.calls.length).to.equal(1);
-          expect(creds.needsRefresh()).to.equal(true);
-        });
-      });
-
-      describe('refresh', function() {
-        it('loads credentials from specified relative URI', function() {
-          var callbackErr = null;
-          process.env['AWS_CONTAINER_CREDENTIALS_RELATIVE_URI'] = '/path';
-          var spy = mockEndpoint(new Date(AWS.util.date.getDate().getTime() + 100000));
-          creds.refresh(function(err) {
-            callbackErr = err;
-          });
-          expect(spy.calls.length).to.equal(1);
-          expect(spy.calls[0].arguments[0])
-            .to.equal('http://169.254.170.2/path');
-          expect(callbackErr).to.be["null"];
-          expect(creds.accessKeyId).to.equal('KEY');
-          expect(creds.secretAccessKey).to.equal('SECRET');
-          expect(creds.sessionToken).to.equal('TOKEN');
-          expect(creds.needsRefresh()).to.equal(false);
-        });
-
-        it('loads credentials from specified full URI', function() {
-          var callbackErr = null;
-          process.env['AWS_CONTAINER_CREDENTIALS_FULL_URI'] = 'http://localhost/get-credentials';
-          var spy = mockEndpoint(new Date(AWS.util.date.getDate().getTime() + 100000));
-          creds.refresh(function(err) {
-            callbackErr = err;
-          });
-          expect(spy.calls.length).to.equal(1);
-          expect(spy.calls[0].arguments[0])
-            .to.equal('http://localhost/get-credentials');
-          expect(callbackErr).to.be["null"];
-          expect(creds.accessKeyId).to.equal('KEY');
-          expect(creds.secretAccessKey).to.equal('SECRET');
-          expect(creds.sessionToken).to.equal('TOKEN');
-          expect(creds.needsRefresh()).to.equal(false);
```

Comment:
Why are we removing these tests?

Reply:
EcsCredentials and RemoteCredentials refer to the same class, but the same set of tests were created for both symbols. I replaced the EcsCredentials tests with a single test that asserts the symbol is the same as RemoteCredentials.

The online diff view is jacked up for me, so the actual changes might not be apparent.

Reply:
I see. I may spend a little more time to review the unit tests changes. The diff is not as clear as it should be.

Reply:
I double-checked the diff in [kaleidoscope](https://www.kaleidoscopeapp.com/) (my preferred diff viewer), and it looks the way I intended, but please let me know if you find anything that doesn't look right.